### PR TITLE
[PM-5237] Clients, Self Hosted: Login - Hide "Create account" when registration disabled

### PR DESCRIPTION
**Pull request by alec-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-5237?atlOrigin=eyJpIjoiYzc4ODNlZjM5Y2RiNDY5Njg5YmU5MjM3YjgwYjZjYTkiLCJwIjoiaiJ9

## 📔 Objective

Shows or hides the "create account" link based on the GlobalSettings configuration.

## 📸 Screenshots

Screenshot 2024-10-31 at 1 16 25 PM
Screenshot 2024-10-31 at 1 16 12 PM

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Comment by alec-livefront** [comment]
@JaredSnider-Bitwarden would you mind taking a look at this and letting me know if I'm on the right track? One piece I haven't been able to work out is "The create account section should react to environment changes if a user changes from an env where open registration is enabled to one where it is disabled" - it seems I have to refresh the page to get it to pick up the latest config.

File: apps/web/src/app/auth/login/login-v1.component.html
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/auth/login/login-v1.component.html:1]: Let's not update any v1 components.

File: libs/common/src/platform/services/server-settings.service.ts
Comment (JaredSnider-Bitwarden) [inline libs/common/src/platform/services/server-settings.service.ts:1]: 💡 : can we add the Default prefix to the service to match our docs, please?

File: libs/common/src/platform/models/domain/server-settings.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface Settings {
+  disableUserRegistration: boolean;
+}
+
```
Comment (JaredSnider-Bitwarden) [inline libs/common/src/platform/models/domain/server-settings.ts:4]: Let's deprecate the interface and just work directly with the `ServerSettings` class

File: libs/common/src/platform/services/server-settings.service.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { map, Observable } from "rxjs";
+
+import { ConfigService } from "../abstractions/config/config.service";
+import { ServerSettings, Settings } from "../models/domain/server-settings";
+
+export class ServerSettingsService {
+  constructor(private configService: ConfigService) {}
+
+  getSettings$(): Observable<Settings> {
+    return this.configService.serverSettings$.pipe(map((settings: ServerSettings) => settings));
```
Comment (JaredSnider-Bitwarden) [inline libs/common/src/platform/services/server-settings.service.ts:10]: Let's clean up this map as it isn't necessary, please

**Review by JaredSnider-Bitwarden — CHANGES_REQUESTED** [review]
Per our call, some notes below!

File: apps/web/src/app/auth/login/login-v1.component.html
Comment (alec-livefront) [reply apps/web/src/app/auth/login/login-v1.component.html:1]: [3493673](https://github.com/bitwarden/clients/pull/11811/commits/34936733971eaa6f11fa58e02bc4d3cd636eedb8)

File: libs/common/src/platform/services/server-settings.service.ts (hunk 2 above)
Comment (alec-livefront) [reply libs/common/src/platform/services/server-settings.service.ts]: [3d1dcf3](https://github.com/bitwarden/clients/pull/11811/commits/3d1dcf3c431b0af9652d46d239ec0a7e867b6bac)

File: libs/common/src/platform/services/server-settings.service.ts
Comment (alec-livefront) [reply libs/common/src/platform/services/server-settings.service.ts:1]: [d63753e](https://github.com/bitwarden/clients/pull/11811/commits/d63753e55f62be9d5f6f91832da9796080227b13)

File: libs/common/src/platform/models/domain/server-settings.ts (hunk 1 above)
Comment (alec-livefront) [reply libs/common/src/platform/models/domain/server-settings.ts]: [a08fd41](https://github.com/bitwarden/clients/pull/11811/commits/a08fd41b51c2b25b50ee5e0080a7cad55a5e7d7b)

File: apps/web/src/app/auth/login/login-v1.component.html [hunk 3]
```diff
@@ -44,8 +44,6 @@
       </a>
     </div>
 
-    <hr />
```
Comment (JaredSnider-Bitwarden) [inline apps/web/src/app/auth/login/login-v1.component.html:47]: This <hr /> seems to have been removed in error?

File: libs/common/src/platform/services/default-server-settings.service.spec.ts
Comment (JaredSnider-Bitwarden) [inline libs/common/src/platform/services/default-server-settings.service.spec.ts:1]: Nice test suite!

**Review by JaredSnider-Bitwarden — CHANGES_REQUESTED** [review]
One tiny clean up item and this is done! Excellent work!

File: apps/web/src/app/auth/login/login-v1.component.html (hunk 3 above)
Comment (alec-livefront) [reply apps/web/src/app/auth/login/login-v1.component.html]: [4434520](https://github.com/bitwarden/clients/pull/11811/commits/4434520e8de6d635d07171c02ce8cc6322d812fe)

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by addisonbeck — APPROVED** [review]

**Pull request merged**